zebra/if_netlink.c

```diff
@@ -1206,8 +1209,12 @@ int netlink_link_change(struct sockaddr_nl *snl, struct nlmsghdr *h,
 		else if (IS_ZEBRA_IF_VXLAN(ifp))
 			zebra_l2_vxlanif_del(ifp);
 
-		if (!IS_ZEBRA_IF_VRF(ifp))
+		if (!IS_ZEBRA_IF_VRF(ifp)) {
 			if_delete_update(ifp);
+			if (IS_ZEBRA_DEBUG_KERNEL)
+				zlog_debug ("delete pseudo interface %s(%u) from zebra", name, ifp->ifindex);
+			if_delete(ifp);
```

Comment:
This is not correct, it will delete all user configuration related to that interface.